src/components/common/BottomsheetComponent.jsx

```diff
@@ -0,0 +1,314 @@
+import { useEffect, useRef } from 'react';
+import { createPortal } from 'react-dom';
+import styled from 'styled-components';
+import { useBottomsheetStore } from '../../stores/useBottomsheetStore';
+
+// 공통 styled-components
+const SHandleBar = styled.div`
+  display: flex;
+  position: relative;
+  width: 100%;
+  height: 1.5rem;
+  justify-content: center;
+  align-items: center;
+  background-color: white;
+  border-top-left-radius: 1rem;
+  border-top-right-radius: 1rem;
+  cursor: grab;
+  touch-action: none;
+  user-select: none;
+
+  &:active {
+    cursor: grabbing;
+  }
+
+  &::before {
+    content: '';
+    width: 2.5rem;
+    height: 0.25rem;
+    background-color: #d1d5db;
+    border-radius: 0.125rem;
+  }
+`;
+
+const SContent = styled.div`
+  width: 100%;
+  background-color: white;
+  overflow-y: auto;
+  transition: height 0.3s cubic-bezier(0.25, 0.46, 0.45, 0.94);
+`;
+
+// 지도 탐색용 바텀시트 (non-modal)
+const SMapBottomsheetLayout = styled.div`
```

Comment:
❗ 바텀시트 bg color는 Layout에서 한 번에 지정해 주세요.
```suggestion
const SMapBottomsheetLayout = styled.div`
  background-color: var(--colors-bg);
```